File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robot/BulkReads.java
package org.firstinspires.ftc.teamcode.robot;

import com.qualcomm.hardware.lynx.LynxModule;
import com.qualcomm.robotcore.hardware.HardwareMap;

import java.util.List;

public class BulkReads {
    private final List<LynxModule> lynxModules;

    public BulkReads(HardwareMap hardwareMap) {
        lynxModules = hardwareMap.getAll(LynxModule.class);

        for (LynxModule hub : lynxModules) {
            hub.setBulkCachingMode(LynxModule.BulkCachingMode.MANUAL);
        }
    }

    public void readAll() {
        for (LynxModule hub : lynxModules) {
            hub.clearBulkCache();
        }
    }
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robot/BulkReads.kt
package org.firstinspires.ftc.teamcode.robot

import com.qualcomm.hardware.lynx.LynxModule
import com.qualcomm.robotcore.hardware.HardwareMap

class BulkReads(hardwareMap: HardwareMap) {
    private val lynxModules: List<LynxModule> = hardwareMap.getAll(LynxModule::class.java)

    init {
        for (hub in lynxModules) {
            hub.bulkCachingMode = LynxModule.BulkCachingMode.MANUAL
        }
    }

    fun readAll() {
        for (hub in lynxModules) {
            hub.clearBulkCache()
        }
    }
}
